### ISSUE TITLE
Progressive mode is failing for new files

### ISSUE THREAD

**Issue by kostyaplis** [ansible/ansible-lint#2363]
##### Summary

When scanning a newly added file with `progressive: true`, `FileNotFoundError` exception is raised because the file doesn't exist in the previous commit
Note: only reproducible when passing the file path as an argument to ansible-lint.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
$ ansible-lint --version                                                                    
ansible-lint 6.5.1 using ansible 2.13.3
```

##### STEPS TO REPRODUCE

1. Add `playbook-test1.yml` to the existing git repository
2. Commit
3. Run ansible-lint in progressive mode 

```console (paste below)
$ ansible-lint --progressive playbook-test1.yml

Preparing worktree (checking out 'old-rev')
HEAD is now at f17ba41 Second commit. Fix 1 violation
Traceback (most recent call last):
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/bin/ansible-lint", line 8, in <module>
    sys.exit(_run_cli_entrypoint())
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/site-packages/ansiblelint/__main__.py", line 281, in _run_cli_entrypoint
    sys.exit(main(sys.argv))
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/site-packages/ansiblelint/__main__.py", line 211, in main
    old_result = _get_matches(rules, options)
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/site-packages/ansiblelint/runner.py", line 221, in _get_matches
    matches.extend(runner.run())
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/site-packages/ansiblelint/runner.py", line 120, in run
    lintable.data
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/site-packages/ansiblelint/file_utils.py", line 294, in data
    self._data = parse_yaml_linenumbers(self)
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/site-packages/ansiblelint/utils.py", line 739, in parse_yaml_linenumbers
    loader = AnsibleLoader(lintable.content, **kwargs)
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/site-packages/ansiblelint/file_utils.py", line 220, in content
    self._populate_content_cache_from_disk()
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/site-packages/ansiblelint/file_utils.py", line 212, in _populate_content_cache_from_disk
    self._content = self.path.resolve().read_text(encoding="utf-8")
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/pathlib.py", line 1132, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "/Users/konstantinplis/.pyenv/versions/3.10.1/lib/python3.10/pathlib.py", line 1117, in open
    return self._accessor.open(self, mode, buffering, encoding, errors,
FileNotFoundError: [Errno 2] No such file or directory: '/Users/konstantinplis/.cache/ansible-compat/f23de3/old-rev/playbook-test1.yml'
```

##### Desired Behavior

Should previous versions of newly added files be treated as an empty string?

##### Actual Behavior

Unhandled exception
